File: Dataset/cpp/uplow.cpp

//  uplow.cpp (c) 1998-2002 Kari Laitinen

#include  <iostream.h>

void  convert_string_uppercase( char given_string[] )
{
   int  character_index  =  0 ;

   while ( given_string[ character_index ]  !=  0 )
   {
      if ( given_string[ character_index ]  >=  'a'  &&
           given_string[ character_index ]  <=  'z'  )
      {
         given_string[ character_index ]  =  
                given_string[ character_index ]  &  0xDF ;
      }

      character_index  ++  ;
   }
}

int main()
{
   char  test_string[]  =  "Bjarne Stroustrup invented C++." ;

   cout << "\n String before conversion:  " << test_string ;

   convert_string_uppercase( test_string ) ;

   cout << "\n String after conversion:   " << test_string ;
}

File: Dataset/cs/Uplow.cs

//  Uplow.cs  (c) 2004 Kari Laitinen

//  www.naturalprogramming.com

using System ;
using System.Text ;  // Class StringBuilder etc.

class Uplow
{
   static string convert_string_uppercase( string given_string )
   {
      StringBuilder modified_string  =  new StringBuilder( given_string ) ;

      int  character_index  =  0 ;

      while ( character_index  <  modified_string.Length )
      {
         if ( modified_string[ character_index ]  >=  'a'  &&
              modified_string[ character_index ]  <=  'z'  )
         {
            modified_string[ character_index ]  =  
               (char)  ( modified_string[ character_index ]  &  0xDF ) ;
         }

         character_index  ++  ;
      }

      return  modified_string.ToString() ;
   }

   static void Main()
   {
      string  test_string  =    "Hejlsberg, Wiltamuth, and Golde"
                             +  " are inventors of C#." ;

      Console.Write( "\n Original string:   "  +  test_string ) ;

      string  uppercase_string  =  convert_string_uppercase( test_string ) ;

      Console.Write( "\n After conversion:  "  +  uppercase_string ) ;

      Console.Write( "\n Better conversion: "
                  +  test_string.ToUpper() ) ;
   }
}






